pkg/descheduler/descheduler.go

```diff
@@ -67,10 +76,92 @@ func Run(rs *options.DeschedulerServer) error {
 
 type strategyFunction func(ctx context.Context, client clientset.Interface, strategy api.DeschedulerStrategy, nodes []*v1.Node, podEvictor *evictions.PodEvictor, getPodsAssignedToNode podutil.GetPodsAssignedToNodeFunc)
 
+func cachedClient(
+	realClient clientset.Interface,
+	podInformer corev1informers.PodInformer,
+	nodeInformer corev1informers.NodeInformer,
+	namespaceInformer corev1informers.NamespaceInformer,
+	priorityClassInformer schedulingv1informers.PriorityClassInformer,
+) (clientset.Interface, error) {
+	fakeClient := fakeclientset.NewSimpleClientset()
```

Comment:
Do we need fakeClient, can't we get away with a local cache of pods just like we do in case of scheduler?

Reply:
I used the fake client so the strategies can stay unchanged. Replacing one client for another. Creating a local cache of pods would require more changes. Perhaps I don't see the full picture. Would you mind describing the approach in more detail? How a strategy is expected to consume the local cache?